File: src/main/java/com/example/demo/HelloController.java
package com.example.demo;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Hooks;
import reactor.core.publisher.Mono;

@RestController
public class HelloController {
    private Logger logger = LoggerFactory.getLogger(HelloController.class);

    public HelloController() {
        Hooks.enableAutomaticContextPropagation();
    }

    @GetMapping("/")
    public Mono<String> index() {
        logger.error("greet");
        return Mono.just("Greetings from Spring Boot!").flatMap((greet) -> {
            logger.error(greet);
            return Mono.just("Greetings from Spring Boot!");
        });
    }

}

File: src/main/kotlin/com/example/demo/HelloController.kt
package com.example.demo

import kotlinx.coroutines.reactor.mono
import org.slf4j.Logger
import org.slf4j.LoggerFactory
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController
import reactor.core.publisher.Hooks
import reactor.core.publisher.Mono


@RestController
class HelloController {
    private val logger: Logger = LoggerFactory.getLogger(HelloController::class.java)

    init {
        Hooks.enableAutomaticContextPropagation()
    }

    @GetMapping("/")
    fun index(): Mono<String> {
        logger.trace("greet")
        return mono {
            val text = "Greetings from Spring Boot!"
            logger.error(text)
            text
        }
    }
}